rest/src/main/scala/org/apache/gearpump/services/RestServices.scala

```diff
@@ -20,59 +20,48 @@ package org.apache.gearpump.services
 
 import akka.actor.{Actor, ActorRef, ActorSystem, Props}
 import akka.io.IO
-import com.gettyimages.spray.swagger._
-import com.wordnik.swagger.model.ApiInfo
+import org.apache.gearpump.cluster.AppMasterToMaster.AppMasterDataDetail
+import org.apache.gearpump.cluster.UserConfig
+import org.apache.gearpump.partitioner.Partitioner
+import org.apache.gearpump.streaming.{AppDescription, TaskDescription, DAG}
+import org.apache.gearpump.util.Graph
 import spray.can._
-import spray.json._
-import spray.routing.HttpService
 
 import scala.concurrent.ExecutionContext
-import scala.reflect.runtime.universe._
 
-class RestServices(masters: ActorRef) extends Actor with HttpService with DefaultJsonProtocol {
-  def actorRefFactory = context
-  implicit val executionContext:ExecutionContext = context.dispatcher
-
-  val appMastersService = new AppMastersService {
-    val master = masters
-    def actorRefFactory = context
-  }
-
-  val appMasterService = new AppMasterService {
-    val master = masters
-    def actorRefFactory = context
-  }
+trait RestServices extends AppMastersService with AppMasterService {
+  val route = appMastersRoute ~ appMasterRoute ~
+    get {
+      pathPrefix("") {
+        pathEndOrSingleSlash {
+          getFromResource("index.html")
+        }
+      }
+    }
+}
 
-  override def preStart(): Unit = {
-    context.actorOf(Props(classOf[AppMastersServiceActor], masters), "appMastersService")
-    context.actorOf(Props(classOf[AppMasterServiceActor], masters), "appMasterService")
+class RestServicesActor(masters: ActorRef) extends Actor with RestServices {
+  val master = masters
+  def actorRefFactory = context
+  implicit val system = context.system
+  implicit val executionContext: ExecutionContext = context.dispatcher
+  def dag(config: Option[UserConfig]): DAG = {
```

Comment:
We could do something like 
        val dag = DAG(_config.getValue[Graph[TaskDescription, Partitioner]](AppDescription.DAG).getOrElse(Graph.empty[TaskDescription, Partitioner]))

since in general UserConfig.getValue[T] should be well behaved.